dockstore-webservice/src/test/java/io/dockstore/webservice/permissions/sam/SamPermissionsImplTest.java

```diff
@@ -490,6 +492,35 @@ public void testSelfDestruct() throws ApiException {
 
     }
 
+    @Test
+    public void testSetPermissionWhenResourceCreatedWithoutAllPolicies() throws ApiException {
+        when(fooWorkflow.getUsers()).thenReturn(new HashSet<>(Arrays.asList(userMock)));
+        final String resourceId = SamConstants.ENCODED_WORKFLOW_PREFIX + FOO_WORKFLOW_NAME;
+        ResourceAndAccessPolicy owner = resourceAndAccessPolicyHelper(resourceId, SamConstants.OWNER_POLICY);
+        when(resourcesApiMock.listResourcesAndPolicies(SamConstants.RESOURCE_TYPE))
+                .thenReturn(Arrays.asList(owner));
+        setupInitializePermissionsMocks(resourceId);
+        samPermissionsImpl.setPermission(userMock, fooWorkflow, new Permission("johndoe@example.com", Role.WRITER));
+        verify(resourcesApiMock, times(1)).overwritePolicy(eq(SamConstants.RESOURCE_TYPE), anyString(), eq(SamConstants.WRITE_POLICY), any());
+    }
+
+    @Test
+    public void testSamPolicyOwnedbyAnother() throws ApiException {
+        when(fooWorkflow.getUsers()).thenReturn(new HashSet<>(Arrays.asList(userMock)));
+        final String resourceId = SamConstants.WORKFLOW_PREFIX + FOO_WORKFLOW_NAME;
+        when(resourcesApiMock.listResourcePolicies(SamConstants.RESOURCE_TYPE, resourceId))
+                .thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "")); // If you don't have permissions, you get a 404
+        doThrow(new ApiException(HttpStatus.SC_CONFLICT, ""))
+                .when(resourcesApiMock)
+                .createResourceWithDefaults(SamConstants.RESOURCE_TYPE, resourceId);
+        try {
+            samPermissionsImpl.setPermission(userMock, fooWorkflow, new Permission("johndoe@example.com", Role.WRITER));
+            Assert.fail("Expected setPermission to throw exception");
+        } catch (CustomWebApplicationException ex) {
+            Assert.assertTrue(ex.getMessage().contains(" 409 "));
```

Comment:
Spacing here intentional?

Reply:
Yes, spacing was spacing intentional, although probably overkill. The `CustomWebApplicationException` concatenates the status code and error into the message, and there are spaces around the `409` when it does so, so I thought it would be slightly extra safe to check for that.